src/NHibernate.Test/NHSpecificTest/GH1712/Entity.cs

```diff
@@ -0,0 +1,13 @@
+﻿using System;
+
+namespace NHibernate.Test.NHSpecificTest.GH1712
+{
+	class GenericEntity<TId> : IEquatable<GenericEntity<TId>> where TId : IEquatable<TId>
+	{
+		public virtual TId Id { get; set; }
+		public virtual bool Equals(GenericEntity<TId> other)
+		{
+			throw new NotImplementedException();
```

Comment:
Doesn't this proof my point that we shall not try to add support for the generic `IEquatable<>`?

Reply:
I do not follow you. What proof do you see there? Is it the fact than someone may implement something else than the equality that will be applied on by NHibernate on the IDs, and so have a discrepancy between linq-to-object and linq-to-nhibernate? And then? That is also the case for `object.Equals`, that is also the case for the `==` operator. If that is your proof, then we should also remove support of `==` on entities in the Linq provider. And many more similar cases...

Reply:
@fredericDelaporte what is the current behavior if there is no `IEquatable<GenericEntity<TId>>` eg if you are using `object.Equals(object)` method?

Reply:
Currently not supported, because, like for `TimeSpan`, this overload registration is lacking. Adding it (`ReflectHelper.GetMethodDefinition<object>(x => x.Equals(default(object))),`) will fix this. Indeed it should be added too.

Reply:
> Currently not supported

No, it's actually supported. Re-Linq/Linq handles this and converts to Expression.Eqaul